### PR TITLE
backend/drm: always perform an atomic test-only commit on wlr_output_test

### PR DESCRIPTION
**Pull request by emersion**
~~Depends on: https://github.com/swaywm/wlroots/pull/2829~~

This allows callers to use wlr_output_test to check whether a mode
can be enabled, for instance.

### CODE REVIEW

**Comment by emersion** [comment]
This is now ready for review.

File: backend/drm/drm.c [hunk 1]
```diff
@@ -473,16 +473,19 @@ static bool drm_connector_test(struct wlr_output *output) {
 		}
 	}
 
-	if ((output->pending.committed & WLR_OUTPUT_STATE_BUFFER) && !conn->backend->parent) {
+	if (conn->backend->parent) {
+		// If we're running as a secondary GPU, we can't perform an atomic
+		// commit without blitting a buffer.
```
Comment (kennylevinsen) [inline backend/drm/drm.c:478]: Hmm, so test commits aren't possible in that configuration?

Comment (emersion) [reply backend/drm/drm.c:478]: We need the buffer which will be used for scan-out to do a test-only commit. So yeah, it's tricky to get one without doing a buffer copy. I'd rather disable any fancy features on secondary GPUs for now (which we don't have any right now, but in the future will be output layers, scaling/cropping, etc).

**Review by kennylevinsen — APPROVED** [review]
LGTM, thanks!

**Pull request merged**